frontend/src/components/Pages/FrontPage/MissionOverview/ScheduleMissionDialog.tsx

```diff
@@ -54,60 +69,83 @@ export const ScheduleMissionDialog = (props: IProps): JSX.Element => {
         timer = setTimeout(() => {
             openPopover()
         }, 300)
-
-        if (sessionStorage.getItem('assetString')) setIsAssetValid(true)
-        else setIsAssetValid(false)
     }
 
     const handleClose = () => {
         clearTimeout(timer)
         closePopover()
     }
 
+    const onClickScheduleMission = () => {
+        setIsScheduleMissionsPressed(true)
+        props.onFrontPageScheduleButtonPress()
+    }
+
     const onChangeEchoMissionSelections = (changes: AutocompleteChanges<string>) => {
         props.onSelectedMissions(changes.selectedItems)
     }
     const onChangeRobotSelection = (changes: AutocompleteChanges<string>) => {
         props.onSelectedRobot(changes.selectedItems[0])
     }
+
     return (
         <>
             <div onPointerEnter={handleHover} onPointerLeave={handleClose} onFocus={openPopover} onBlur={handleClose}>
                 <Button
                     onClick={() => {
-                        setIsDialogOpen(true)
+                        onClickScheduleMission()
                     }}
                     disabled={props.frontPageScheduleButtonDisabled}
                     ref={anchorRef}
                 >
-                    <Icon name={Icons.Add} size={16} />
-                    {Text('Add mission')}
+                    {!props.isLoadingEchoMissions && (
+                        <>
+                            <Icon name={Icons.Add} size={16} />
+                            {Text('Add mission')}
+                        </>
+                    )}
+                    {props.isLoadingEchoMissions && (
+                        <>
+                            <Icon name={Icons.Wait} size={16} />
+                            {Text('Loading')}..
+                        </>
+                    )}
                 </Button>
             </div>
 
             <Popover
                 anchorEl={anchorRef.current}
                 onClose={handleClose}
-                open={isPopoverOpen && !isAssetValid}
+                open={isPopoverOpen && asset === ''}
                 placement="top"
             >
                 <Popover.Content>
                     <Typography variant="body_short">{Text('Please select asset')}</Typography>
                 </Popover.Content>
             </Popover>
 
-            <Popover
-                anchorEl={anchorRef.current}
-                onClose={handleClose}
-                open={isPopoverOpen && isAssetValid}
-                placement="top"
-            >
-                <Popover.Content>
-                    <Typography variant="body_short">
-                        {Text('This asset has no missions - Please create mission')}
-                    </Typography>
-                </Popover.Content>
-            </Popover>
+            <StyledMissionDialog>
+                <Dialog open={isNoEchoMissionsDialogOpen} isDismissable>
```

Comment:
Very confusing name "isNoEchoMissionsDialogOpen"?

Reply:
Changed:
isDialogOpen => isScheduleMissionDialogOpen
isNoEchoMissionsDialogOpen => isEmptyEchoMissionsDialogOpen

is that more clear?